app/process/handler/consensus-message/consensus-message.go

```diff
@@ -1,3 +1,110 @@
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/message"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	validatorproto "github.com/limechain/hedera-eth-bridge-validator/proto"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+type ConsensusMessageHandler struct {
+	repository     repositories.MessageRepository
+	validAddresses []string
+}
+
+func NewConsensusMessageHandler(r repositories.MessageRepository) *ConsensusMessageHandler {
+	return &ConsensusMessageHandler{
+		repository:     r,
+		validAddresses: config.LoadConfig().Hedera.Handler.ConsensusMessage.Addresses,
+	}
+}
+
+func (cmh ConsensusMessageHandler) Handle(payload []byte) {
+	err := cmh.handlePayload(payload)
+	if err != nil {
+		log.Errorf("Error - could not handle payload: [%s]", err)
+		return
+	}
+}
+
+func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
+	m := &validatorproto.TopicSignatureMessage{}
+	err := proto.Unmarshal(payload, m)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to unmarshal topic signature message. - [%s]", err))
+	}
+
+	ctm := &validatorproto.CryptoTransferMessage{
+		TransactionId: m.TransactionId,
+		EthAddress:    m.EthAddress,
+		Amount:        m.Amount,
+		Fee:           m.Fee,
+	}
+
+	log.Printf("[%s] signature", m.GetSignature())
+
+	decodedSig, err := hex.DecodeString(m.GetSignature())
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to decode signature. - [%s]", err))
+	}
+
+	hash := crypto.Keccak256([]byte(fmt.Sprintf("%s-%s-%d-%s", ctm.TransactionId, ctm.EthAddress, ctm.Amount, ctm.Fee)))
+
+	log.Printf("[%s] hex hash", hex.EncodeToString(hash))
+
+	key, err := crypto.Ecrecover(hash, decodedSig)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to recover public key. - [%s]", err))
+	}
+
+	pubKey, err := crypto.UnmarshalPubkey(key)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to unmarshal public key. - [%s]", err))
+	}
+
+	address := crypto.PubkeyToAddress(*pubKey)
+
+	if !cmh.isValidAddress(address.String()) {
+		return errors.New(fmt.Sprintf("Address is not valid - [%s]", address.String()))
+	}
+
+	messages, err := cmh.repository.GetByTxIdAndSignature(m.TransactionId, m.Signature)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to retrieve messages for TxId [%s], with signature [%s]. - [%s]", m.TransactionId, m.Signature, err))
+	}
+
+	if len(messages) > 0 {
+		return errors.New(fmt.Sprintf("Duplicated Transaction Id and Signature - [%s]-[%s]", m.TransactionId, m.Signature))
+	}
+
+	err = cmh.repository.Create(&message.TransactionMessage{
+		TransactionId: m.TransactionId,
+		EthAddress:    m.EthAddress,
+		Amount:        m.Amount,
+		Fee:           m.Fee,
+		Signature:     m.Signature,
+		Hash:          hex.EncodeToString(hash),
+	})
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not add Transaction Message with Transaction Id and Signature - [%s]-[%s]", m.TransactionId, m.Signature))
+	}
+
+	fmt.Println("Success.")
```

Comment:
We must log the TX ID as-well.
Something like:
`Successfully verified and persisted TX with ID [ID]`